### ISSUE TITLE
SyntaxError on `yo keystone`

### ISSUE THREAD

**Issue by misterdai** [keystonejs/generator-keystone#189]
Hi, was hoping to test out keystone for a site but ran into an issue with running the generator.

* generator-keystone `3.1.6`
* Yo `1.8.1`
* Node `4.4.4`

All works fine with answering the questions but as soon as I answer the `include extra code comments` (answer does not affect outcome), I get the following `SyntaxError`.

```
? ------------------------------------------------
    Finally, would you like to include extra code comments in
    your project? If you're new to Keystone, these may be helpful. No
/usr/local/lib/node_modules/generator-keystone/node_modules/yeoman-generator/node_modules/lodash/dist/lodash.js:6309
        throw e;
        ^

SyntaxError: Unexpected token =
    at Function (native)
    at Function.template (/usr/local/lib/node_modules/generator-keystone/node_modules/yeoman-generator/node_modules/lodash/dist/lodash.js:6306:22)
    at KeystoneGenerator.underscore [as _engine] (/usr/local/lib/node_modules/generator-keystone/node_modules/yeoman-generator/lib/util/engines.js:31:14)
    at KeystoneGenerator.engine (/usr/local/lib/node_modules/generator-keystone/node_modules/yeoman-generator/lib/actions/actions.js:314:10)
    at KeystoneGenerator.template (/usr/local/lib/node_modules/generator-keystone/node_modules/yeoman-generator/lib/actions/actions.js:289:15)
    at KeystoneGenerator.project (/usr/local/lib/node_modules/generator-keystone/app/index.js:289:7)
    at /usr/local/lib/node_modules/generator-keystone/node_modules/yeoman-generator/lib/base.js:395:14
    at processImmediate [as _immediateCallback] (timers.js:383:17)
```

**Comment by mxstbr** [comment]
The current version is `0.3.17`, can you try running `npm install -g generator-keystone` to update the generator and seeing if the bug still exists?

**Comment by misterdai** [comment]
I'm only seeing `0.3.15` on npm https://www.npmjs.com/package/generator-keystone

```
npm ERR! No compatible version found: generator-keystone@0.3.17
npm ERR! Valid install targets:
npm ERR! 0.3.15, 0.3.14, 0.3.13, 0.3.12, .....
```

**Comment by mxstbr** [comment]
Ah yeah I was overeager, 0.3.15 is the latest one, I got a bit confused with the main keystone module!

**Comment by mxstbr** [comment]
@r3wt and @luvmex have confirmed this is indeed a bug [here](https://github.com/keystonejs/keystone/issues/2820), would appreciate somebody looking into it!

**Comment by r3wt** [comment]
the error comes from `_package.json` it appears

**Comment by morenoh149** [comment]
not seeing anything obvious here https://github.com/keystonejs/generator-keystone/blob/master/app/templates/_package.json

**Comment by misterdai** [comment]
Kind of wish lodash templates were easier to debug, but seems an additional equals sign crept in.  I've created the above linked pull request to remove it.

**Comment by troystarwalt** [comment]
This won't be helpful, but I'm experiencing this same problem with a fresh install. Happy to provide more info if needed.

**Comment by r3wt** [comment]
@troystarwalt its already fixed, and should land soon. see @misterdai 's open PR

**Comment by troystarwalt** [comment]
Just saw that. Thank you!

**Comment by mxstbr** [comment]
Thanks @misterdai, that was quick, merged, will release asap!

**Comment by mxstbr** [comment]
Released as `v0.3.16`!

**Comment by Mariano89** [comment]
Hi guys, I'm still experiencing this error. I believe mine might be slightly different. The error points to Unexpected token 'else' instead of '='. Currently running v.0.3.16. 
Here is my error:

**_/usr/local/lib/node_modules/generator-keystone/node_modules/yeoman-generator/node_modules/lodash/dist/lodash.js:6309
        throw e;
        ^

SyntaxError: Unexpected token else
    at Function (native)
    at Function.template (/usr/local/lib/node_modules/generator-keystone/node_modules/yeoman-generator/node_modules/lodash/dist/lodash.js:6306:22)
    at KeystoneGenerator.underscore [as _engine] (/usr/local/lib/node_modules/generator-keystone/node_modules/yeoman-generator/lib/util/engines.js:31:14)
    at KeystoneGenerator.engine (/usr/local/lib/node_modules/generator-keystone/node_modules/yeoman-generator/lib/actions/actions.js:314:10)
    at KeystoneGenerator.template (/usr/local/lib/node_modules/generator-keystone/node_modules/yeoman-generator/lib/actions/actions.js:289:15)
    at KeystoneGenerator.tasks (/usr/local/lib/node_modules/generator-keystone/app/index.js:306:8)
    at /usr/local/lib/node_modules/generator-keystone/node_modules/yeoman-generator/lib/base.js:395:14
    at processImmediate [as _immediateCallback] (timers.js:383:17)_**

Thanks!

**Comment by r3wt** [comment]
@mxstbr looks like coming from https://github.com/keystonejs/generator-keystone/blob/master/app/templates/_Gruntfile.js#L34 this little bad boy(`<% } %>`) should not be there.

**Comment by morenoh149** [comment]
@r3wt I'd be wary of assuming what a contributor did or did not do. People make mistakes.

Perhaps we should begin testing this command line tool https://blog.engineyard.com/2014/bats-test-command-line-tools

cc @webteckie

**Comment by troystarwalt** [comment]
If it helps, I was able to update to the newest version and had no problems running the generator on my end.

**Comment by r3wt** [comment]
> @r3wt I'd be wary of assuming what a contributor did or did not do. People make mistakes.

@morenoh149 I'm not sure you can classify it as an assumption, since running the code results in an error. Agree to disagree on that point, and my post was not to call anyone out specifically (Full disclosure, i didn't test my PR's either), but rather to highlight the need for a review process, to prevent this sort of thing from happening again. I'm sorry if i hurt anyone's feelings, that was certainly not my intention...  

Re the Engineyard blog posts,thats a good find. Wonder if we could simulate the dialog process or mock it in someway. 

```
#!/usr/bin/env bats
@test "Check that generator-keystone runs successfully" {
    // run tests here.
    [ $status = 0 ]
}
```

**Comment by morenoh149** [comment]
I think if you `sleep 10` between cli entries and check that the filesystem has certain files, that'd be a complete test.

**Comment by mcclaskiem** [comment]
@Mariano89 I added this PR which should fix your issue: https://github.com/keystonejs/generator-keystone/pull/195

**Comment by mcclaskiem** [comment]
#195 @Mariano89 this should fix your issue

**Comment by misterdai** [comment]
@r3wt just to cover myself ;) I didn't select Grunt when I fixed the other issue with `_package.json` so I'll pass the buck to `git blame` on that one :)

**Comment by younesherlock** [comment]
I have chosen `gulp` instead of `grunt` and it generated the project.

**Comment by mcclaskiem** [comment]
@younesherlock the PR I just had merged should have grunt working fine if you were having issues before I would try #195

**Comment by julianovidal** [comment]
Just got the same issue. =(

**Comment by FranckLantres** [comment]
Hi,

Same issue with the following releases :
- generator-keystone 3.7.3
- node 6.2.1
- yo 1.8.4

**Comment by Gutsie** [comment]
Still having this problem if generating the project with grunt

**Comment by mcclaskiem** [comment]
You need to checkout the latest master branch for the fix I had merged in #195. A new tag hasn't been released in 30 days so NPM won't include the changes to fix grunt.
